### PR TITLE
Updated test script, docs, golint suggestions

### PR DESCRIPTION
**Pull request by sdemos**
I added a test script that runs go vet and golint, made testing run on all packages, and deleted the seemingly unused cover script. Then I fixed all the issues that golint complained about, mostly documentation of exported entities. 

The test script will print out golint recommendations, but it does not enforce that they are all met (see https://github.com/golang/lint#purpose).

### CODE REVIEW

File: lock/client.go [hunk 1]
```diff
@@ -14,7 +14,8 @@
 
 package lock
 
-type LockClient interface {
+// Client is a generic interface for a lock
+type Client interface {
```
Comment (dghubble) [inline lock/client.go]: I don't think we should change the name here. There may be importers we break.

Comment (sdemos) [inline lock/client.go]: I agree. I removed the hard requirement of passing golint from the test script. Now it just prints out the recommendations, but passes either way.

Comment (dghubble) [inline lock/client.go]: Huh? Don't you just need the comment to be "LockClient is blah"?

Comment (sdemos) [inline lock/client.go]: golint complains about "stuttering" which is the problem that I was trying to fix (callers outside the lock package refer to this as `lock.LockClient`)

Comment (dghubble) [inline lock/client.go]: ah, ok

**Comment by squeed** [comment]
For the record, `golint` is a sub-par piece of software, and gating all commits on `golint` has proven to be fairly disruptive in practice. I'd suggest integrating `gometalinter` instead, as it supports things like ignore directives.

**Pull request merged**